packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -134,48 +97,18 @@ export const Stackbox = () => {
   const { chain } = useNetwork();
   const { switchNetwork } = useSwitchNetwork({
     onSuccess(data) {
-      setChainId(data.id);
+      handleChainIdChange(data.id);
     },
   });
   const { address } = useAccount();
 
-  const getDefaultParsedToken = (tokenDirection: "to" | "from") =>
-    createParser({
-      parse: (address: string) => getTokenFromList(address),
-      serialize: (token) => token?.address || "",
-    }).withDefault(
-      chain?.id && !chain?.unsupported
-        ? DEFAULT_TOKENS_BY_CHAIN[chain.id][tokenDirection]
-        : DEFAULT_TOKENS_BY_CHAIN[ChainId.GNOSIS][tokenDirection]
-    );
-
-  const [fromToken, setFromToken] = useQueryState<TokenWithBalance>(
-    "fromToken",
-    getDefaultParsedToken("from")
-  );
-  const [toToken, setToToken] = useQueryState<TokenWithBalance>(
-    "toToken",
-    getDefaultParsedToken("to")
-  );
-  const [tokenAmount, setTokenAmount] = useQueryState(
-    "tokenAmount",
-    parseAsString.withDefault("")
-  );
-  const [frequency, setFrequency] = useQueryState(
-    "frequency",
-    parseAsStringEnum<FREQUENCY_OPTIONS>(
-      Object.values(FREQUENCY_OPTIONS)
-    ).withDefault(FREQUENCY_OPTIONS.hour)
-  );
-  const [startDateTime, setStartDateTime] = useQueryState(
-    "startDate",
-    startDateParseAsTimestamp.withDefault(new Date(Date.now()))
-  );
-  const [endDateTime, setEndDateTime] = useQueryState(
-    "endDate",
-    parseAsTimestamp.withDefault(new Date(endDateByFrequency[frequency]))
-  );
-  const [chainId, setChainId] = useQueryState("chainId", parseAsInteger);
+  const [fromToken, setFromToken] = stackboxFormState.fromTokenState;
+  const [toToken, setToToken] = stackboxFormState.toTokenState;
+  const [tokenAmount, setTokenAmount] = stackboxFormState.tokenAmountState;
+  const [frequency, setFrequency] = stackboxFormState.frequencyState;
+  const [startDateTime, setStartDateTime] = stackboxFormState.startDateState;
+  const [endDateTime, setEndDateTime] = stackboxFormState.endDateState;
+  const [chainId, handleChainIdChange] = stackboxFormState.chainIdState;
```

Comment:
To be consistent, let's use `setChainId` instead of handleChainIdChange
```suggestion
  const [chainId, setChainID] = stackboxFormState.chainIdState;
```

Reply:
As defined in an offline chat, will rename this function into `setChainId` as it would be an expected behavior to reset the form values when changing the chain.